hamcrest-core/src/main/java/org/hamcrest/core/Every.java

```diff
@@ -14,19 +15,41 @@ public Every(Matcher<? super T> matcher) {
 
     @Override
     public boolean matchesSafely(Iterable<? extends T> collection, Description mismatchDescription) {
+        List<SelfDescribingText> mismatches = new ArrayList<SelfDescribingText>();
         for (T t : collection) {
             if (!matcher.matches(t)) {
+                mismatches.add(mismatchString(t));
+            }
+        }
+
+        switch (mismatches.size()) {
+            case 0:
+                return true;
+
+            case 1:
                 mismatchDescription.appendText("an item ");
-                matcher.describeMismatch(t, mismatchDescription);
+                mismatchDescription.appendDescriptionOf(mismatches.get(0));
+                return false;
+
+            default:
+                mismatchDescription
+                        .appendText("a collection with ")
+                        .appendValue(mismatches.size())
+                        .appendText(" mismatches: ")
+                        .appendList("[", ", ", "]", mismatches);
                 return false;
-            }
         }
-        return true;
+    }
+
+    private SelfDescribingText mismatchString(T t) {
+        Description desc = new StringDescription();
+        matcher.describeMismatch(t, desc);
+        return new SelfDescribingText(desc.toString());
     }
 
     @Override
     public void describeTo(Description description) {
-        description.appendText("every item is ").appendDescriptionOf(matcher);
+        description.appendText("a collection in which every item is ").appendDescriptionOf(matcher);
```

Comment:
As the `Every` matcher can also be used for `Iterables`: should the text talk about "iterables"?